resource/graph_node.go

```diff
@@ -111,6 +133,24 @@ func (w *GraphNode) Resource() (Resource, error) {
 	return w.current, nil
 }
 
+// WithState contains a resource along with lifecycle state information.
+type WithState struct {
+	Resource  Resource
+	State     NodeState
+	UpdatedAt time.Time
+}
+
+// ResourceWithState returns the underlying resource along with its current state.
+func (w *GraphNode) ResourceWithState() WithState {
```

Comment:
This is odd API to me. Why not have a `State()` method and a `TransitionedAt()` method? I suppose I'm confused why we're returning what is effectively a tuple instead of having separate methods, and why `transitionedAt` became `UpdatedAt`.

Reply:
> tuple instead of having separate methods

I bundled them together to avoid consistency issues related to accessing one field then the other when an update happens in between. my thought is that if you are getting a state you generally want update timestamp as well. In retrospect, maybe the field should just hold a struct then.

> why transitionedAt became UpdatedAt

I wanted to name this field `updated_at` initially but there is another field with the same name that is used for testing.